binary_search_tree/binary_search_tree.py

```diff
@@ -5,10 +5,42 @@ def __init__(self, value):
     self.right = None
 
   def insert(self, value):
+    if value < self.value:
+      if self.left is None:
+        self.left = BinarySearchTree(value)
+        return
+      else:
+       self = self.left
```

Comment:
That's a pretty elegant recursive solution.

I like how you just walked down self in the direction the node needed to be inserted.